### DISCUSSION TITLE
Which markup language can be used in Docstrings to format the documentation-hover pop-up?

### DISCUSSION THREAD

**Discussion by Sorontik** [microsoft/pylance-release#7053] [Q&A]
I am very confused.  
I've spent several hours by now, trying to find out how i can format my docstrings properly, but i couldn't find a concise answer anywhere.

I found a few posts that suggest that either markdown or reST is supported or was at some point used.
I also found indications that sometimes one is converted to the other, 

![grafik](https://github.com/user-attachments/assets/89021309-0d11-4106-82d8-750364119936)

As you can see in the attached screenshot, many markdown features like headings, lists and horizontal lines are working, 
but inline formattings like `*italic*` or `**bold**` do not work.
These should both be valid markdown and valid reST - so i expected them to work either way, but apparently they don't.

Markdown style `~~strikethrough~~` and appending two spaces at the end of a line for a line break also don't work.

I suspect this might be due to issues like #627 , #692 or #716, where users complained that certain character sequences were interpreted as formatting when they weren't intended to be.

I also tried to enable the experimental support for reST in the settings - which wreaked havoc - the layouts were all destroyed, seemingly random colors and other formattings were inserted and the behaviour became erratic, like inserting or removing one element completely changed how other were rendered...

So how can I format my docstrings in a way that is both readable in source code and diplayed nicely in the VSC Editor Hover popup?

**Comment by GuePardo0** [comment]
I've also spent a few hours by now trying to undestand how this works at all. Sometimes, adding a single line break changes how everything that I've written in my docstring is formatted. Sometimes it makes all the text appear in one line, sometimes it colors the text like code, sometimes html tags like <**br> work, sometimes they don't... Sometimes, a random </**ul> appears at the end of a part of the text for some reason.
I'm so confused.

** if I dont put this, it actually places that tag in my comment and it becomes invisible
